python/sidecar/typing.py

```diff
@@ -206,6 +206,30 @@ class SearchResponse(RefStudioModel):
     results: list[S2SearchResult]
 
 
+class FileEntryBase(RefStudioModel):
+    name: str
+    path: str
+
+
+class FileEntry(FileEntryBase):
+    file_extension: str
+
+
+class FolderEntry(FileEntryBase):
+    children: list[FileEntry]
+
+
+class ProjectDetails(RefStudioModel):
+    id: str
+    name: str
+    path: str
+
+
+class ProjectFileTreeResponse(RefStudioModel):
```

Comment:
nit: I think that we only use *Response types for the API output. This type is only used now in `projects.py`